assets/js/googlesitekit/api/index.test.js

```diff
@@ -436,6 +479,47 @@ describe( 'googlesitekit.api', () => {
 			);
 			expect( cacheData.cacheHit ).toEqual( false );
 		} );
+
+		it( 'should add tracking info to the data layer when an error is returned on set', async () => {
+			let pushArgs;
+			const dataLayer = {
+				[ DATA_LAYER ]: {
+					push: ( ...args ) => pushArgs = args,
+				},
+			};
+			const config = {
+				trackingEnabled: true,
+			};
+			const { trackEvent } = createTracking( config, dataLayer );
+
+			// Replace the trackEvent implementation to use our version with the mocked dataLayer.
+			const trackEventSpy = jest.spyOn( Tracking, 'trackEvent' ).mockImplementation( trackEvent );
```

Comment:
Let's use the same approach for this as we have in the `dataAPI` tests for this and make assertions against a `dataLayerPushSpy`.

Reply:
👍